benches/bench.rs

```diff
@@ -10,12 +10,14 @@ pub fn bench_from_nibbles(c: &mut Criterion) {
     let mut group = c.benchmark_group("from_nibbles");
 
     for size in SIZE_NIBBLES {
-        let nibbles_data = generate_nibbles(size);
-
         group.throughput(Throughput::Elements(size as u64));
 
-        group.bench_with_input(BenchmarkId::from_parameter(size), &nibbles_data, |b, data| {
-            b.iter(|| Nibbles::from_nibbles(black_box(data)))
+        group.bench_function(BenchmarkId::from_parameter(size), |b| {
+            b.iter_batched(
+                || generate_nibbles_random(size),
```

Comment:
makes sense but this will only work locally; codspeed only uses 1 input regardless of the function used. See ruint benchmarks for a wrapper to run multiple times on codspeed